### PR TITLE
Fix scatter lowering when src is a Number

### PR DESCRIPTION
**Pull request by isuruf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #129106
* #129097
* __->__ #129096

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/lowering.py [hunk 1]
```diff
@@ -3272,18 +3272,23 @@ def scatter_reduce_(self, dim: int, index, src, reduce, *, include_self: bool =
...
-        reduce=reduce,
-        include_self=include_self,
-    )
 
-    if fallback_result:
-        return fallback_result
+    if isinstance(src, Number):
+        src = full_like(self, src)
+
+    if reduce is not None:
```
Comment (peterbell10) [inline torch/_inductor/lowering.py:3279]: Should this check be in `use_scatter_fallback`?

Comment (isuruf) [reply torch/_inductor/lowering.py:3279]: It depends on the op passed to `scatter_fallback`.

Comment (peterbell10) [reply torch/_inductor/lowering.py:3279]: `scatter_fallback` passes the `op_overload` argument to `use_scatter_fallback`, so I don't see why that's an issue?

**Review by peterbell10 — APPROVED** [review]

File: torch/_inductor/lowering.py (hunk 1 above)
Comment (isuruf) [reply torch/_inductor/lowering.py]: Right. Thanks for the review.

**Pull request closed without merging**